game-core/src/main/java/games/strategy/engine/framework/startup/mc/ClientModel.java

```diff
@@ -268,18 +266,17 @@ private IServerStartupRemote getServerStartup() {
    * Resets stats and nulls out references, keeps chat alive.
    */
   public void cancel() {
-    if (messenger == null) {
-      return;
-    }
-    objectStreamFactory.setData(null);
-    messenger.shutDown();
-    chatPanel.setChat(null);
     gameSelectorModel.setGameData(gameDataOnStartup);
     gameSelectorModel.setCanSelect(true);
-    hostIsHeadlessBot = false;
-    gameSelectorModel.setIsHostHeadlessBot(false);
-    gameSelectorModel.setClientModelForHostBots(null);
-    messenger.removeErrorListener(this);
+    if (messenger != null) {
+      messenger.shutDown();
+      messenger.removeErrorListener(this);
+      objectStreamFactory.setData(null);
+      chatPanel.setChat(null);
+      hostIsHeadlessBot = false;
+      gameSelectorModel.setIsHostHeadlessBot(false);
+      gameSelectorModel.setClientModelForHostBots(null);
+    }
```

Comment:
I'm not 100% sure which one of those should get set, so in order to mostly keep the old behaviour I decided only to use 2 method calls.